### PR TITLE
Add Snapshot database Backup/Restore

### PR DESCRIPTION
**Pull request by cwrw**
### Context

Add a new snapshot database where production data can be restored and refreshed via Github workflow.

- Ticket: https://dfedigital.atlassian.net/jira/software/projects/CPDLP/boards/87?selectedIssue=CPDLP-2535

### Changes proposed in this pull request
- Add new snapshot database that will only be in production. This will need to be run manually before pr is merged
- Add workflow which backs up the prod database on demand and uploads to Github artifact 
- Add a restore step which uses `konduit` and connects to another database using `-d` and where the keyvault secret holding the db URL will already be populated. This will restore the snapshot database from the uploaded artifact

### Guidance to review
This was tested on staging and seems to work with the `konduit` commands, which also allows us to connect to the snapshot locally

### CODE REVIEW

File: .github/actions/backup-database/action.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      uses: azure/setup-kubectl@v3
+
+    - name: Install konduit
+      shell: bash
+      run: make install-konduit
+
+    - name: Backup database
+      shell: bash
+      run: |
+        bin/konduit.sh cpd-ecf-${{ inputs.environment }}-web -- pg_dump -E utf8 --compress=1 --clean --if-exists --no-privileges --no-owner --verbose -f backup-${{ inputs.environment }}.sql.gz
```
Comment (ethax-ross) [inline .github/actions/backup-database/action.yml:38]: Should we be concerned about storing production data outside of Azure/in GitHub artifacts, albeit temporarily?

**Review by ethax-ross — APPROVED** [review]
Looks great! 💯

File: .github/actions/backup-database/action.yml (hunk 1 above)
Comment (saliceti) [reply .github/actions/backup-database/action.yml:38]: Could we combine backup and restore in the same job? So we don't need to store anything

Comment (saliceti) [reply .github/actions/backup-database/action.yml:38]: Or siimply store it in azure storage, such as: https://github.com/DFE-Digital/register-trainee-teachers/blob/main/.github/workflows/database-backup.yml

Comment (cwrw) [reply .github/actions/backup-database/action.yml]: amended now and combined under one job, if you can both have another look, thanks

Comment (ethax-ross) [reply .github/actions/backup-database/action.yml]: Looks good to me 👍

**Review by saliceti — APPROVED** [review]

**Pull request merged**